include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -283,34 +307,11 @@ SECTIONS
 	}
 #endif
 
-#include <app_data_alignment.ld>
```

Comment:
With that change done, I guess you should also:
- git rm include/arch/arm/cortex_m/scripts/app_data_alignment.ld
- remove the 3 lines: https://github.com/zephyrproject-rtos/zephyr/blob/master/CMakeLists.txt#L1080
- change the condition https://github.com/zephyrproject-rtos/zephyr/blob/master/CMakeLists.txt#L1113 to
`if(CONFIG_ARC and CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT AND CONFIG_APPLICATION_MEMORY)`

Reply:
> git rm include/arch/arm/cortex_m/scripts/app_data_alignment.ld

Is it guaranteed that the script is not used by other architectures, e.g. ARC?

Reply:
This file is in an ARM specific directory, and I really doubt that the ARC include paths contains an ARM directory.